controllers/secretagentconfiguration_controller.go

```diff
@@ -120,22 +121,29 @@ func (reconciler *SecretAgentConfigurationReconciler) Reconcile(req ctrl.Request
 	}
 
 	//end old main.go
-
-	k8sSecretList := k8ssecrets.GenerateSecretAPIObjects(instance.Spec.Secrets)
-
 	labelsForSecretAgent := func(name string) map[string]string {
 		return map[string]string{"managed-by-secret-agent": "true", "secret-agent-configuration-name": name}
 	}
 
 	if instance.Spec.AppConfig.CreateKubernetesObjects {
-		for _, secret := range k8sSecretList {
-			// Set SecretAgentConfiguration instance as the owner and controller of the secret
-			if err := ctrl.SetControllerReference(&instance, secret, reconciler.Scheme); err != nil {
+		var operation *string
```

Comment:
do we really need a pointer to a string here?

Reply:
I think is nicer to compare against nil. i.e. `if operation != nil {`
rather than `if len(operation) > 0 {` if I were to return a string 🤷

Reply:
changed to string